pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -0,0 +1,79 @@
+# optimization_utils.py
+from typing import List
+
+import numpy as np
+from pandas import DataFrame
+from xarray import DataArray
+
+from pymc_marketing.mmm.utils import CurveCalculator
+
+
+def cost_function(
+    budget_allocations: List[float],
+    df: DataFrame,
+    data: DataFrame,
+    channel_contributions: DataArray,
+) -> float:
+    """
+    Compute the total collaboration of budget allocation across various channels.
+
+    Parameters
+    ----------
+    budget_allocations : list of float
+        The budget allocated to each channel.
+    df : pandas DataFrame
+        The DataFrame containing data about each channel.
+    data : pandas DataFrame
+        The DataFrame containing the x-values for the quadratic fit.
+    channel_contributions : xarray DataArray
+        The channel contributions.
+
+    Returns
+    -------
+    float
+        The total collaboration of the budget allocations.
+
+    Notes
+    -----
+    This function calculates the total collaboration of budget allocations for each channel based on a quadratic curve fitted to the channel data.
+    The total collaboration is computed as the sum of the polynomial evaluation at the allocation for each channel.
```

Comment:
You can add these remarks at the top of the doc strings and not as notes.

Reply:
The word collaboration is confusing. Can you change it please? Maybe "contribution"?